pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -537,7 +539,80 @@ func (dr *DashboardServiceImpl) GetDashboards(ctx context.Context, query *dashbo
 	return dr.dashboardStore.GetDashboards(ctx, query)
 }
 
+func (dr *DashboardServiceImpl) GetUserSharedDashboards(ctx context.Context, user identity.Requester) ([]*dashboards.Dashboard, error) {
+	return dr.getUserSharedDashboards(ctx, user)
+}
+
+func (dr *DashboardServiceImpl) getUserSharedDashboards(ctx context.Context, user identity.Requester) ([]*dashboards.Dashboard, error) {
+	permissions := user.GetPermissions()
+	dashboardPermissions := permissions["dashboards:read"]
+	sharedDashboards := make([]*dashboards.Dashboard, 0)
+	dashboardUids := make([]string, 0)
+	for _, p := range dashboardPermissions {
+		if dashboardUid, found := strings.CutPrefix(p, "dashboards:uid:"); found {
+			if !slices.Contains(dashboardUids, dashboardUid) {
+				dashboardUids = append(dashboardUids, dashboardUid)
+			}
+		}
+	}
+
+	if len(dashboardUids) == 0 {
+		return sharedDashboards, nil
+	}
+
+	dashboardsQuery := &dashboards.GetDashboardsQuery{
+		DashboardUIDs: dashboardUids,
+		OrgID:         user.GetOrgID(),
+	}
+	sharedDashboards, err := dr.dashboardStore.GetDashboards(ctx, dashboardsQuery)
+	if err != nil {
+		return nil, err
+	}
+	return dr.filterUserSharedDashboards(ctx, user, sharedDashboards)
+}
+
+// filterUserSharedDashboards filter dashboards directly assigned to user, but not located in folders with view permissions
+func (dr *DashboardServiceImpl) filterUserSharedDashboards(ctx context.Context, user identity.Requester, userDashboards []*dashboards.Dashboard) ([]*dashboards.Dashboard, error) {
+	filteredDashboards := make([]*dashboards.Dashboard, 0)
+	for _, dashboard := range userDashboards {
+		// Filter out dashboards if user has access to parent folder
+		g, err := guardian.NewByUID(ctx, dashboard.FolderUID, user.GetOrgID(), user)
```

Comment:
we need to check if `dashboard.FolderUID` is empty otherwise the request fails with:
```
logger=context userId=2 orgId=1 uname=<redacted> t=2023-11-23T16:37:38.724442+02:00 level=info msg="Search failed" error="[guardian.dashboardNotFound] failed to check view permissions for dashboard" remote_addr=[::1] traceID=
```